salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,221 @@
+import os.path
+import yaml
+
+from parameterized import param, parameterized
+
+from kubernetes.client.rest import ApiException
+
+from salt.exceptions import CommandExecutionError
+
+from salttesting.mixins import LoaderModuleMockMixin
+from salttesting.unit import TestCase
+from salttesting.mock import MagicMock, mock_open, patch
+from salttesting.helpers import ForceImportErrorOn
+
+import metalk8s_kubernetes_utils
+
+
+YAML_TESTS_FILE = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "files", "test_metalk8s_kubernetes_utils.yaml"
+)
+with open(YAML_TESTS_FILE) as fd:
+    YAML_TESTS_CASES = yaml.safe_load(fd)
+
+
+class Metalk8sKubernetesUtilsTestCase(TestCase, LoaderModuleMockMixin):
+    """
+    TestCase for `metalk8s_kubernetes_utils` module
+    """
+    loader_module = metalk8s_kubernetes_utils
+
+    def test_virtual_success(self):
+        """
+        Tests the return of `__virtual__` function, success
+        """
+        reload(metalk8s_kubernetes_utils)
+        self.assertEqual(
+            metalk8s_kubernetes_utils.__virtual__(),
+            'metalk8s_kubernetes'
+        )
+
+    def test_virtual_fail_import(self):
+        """
+        Tests the return of `__virtual__` function, fail import
+        """
+        with ForceImportErrorOn(
+            "kubernetes.client",
+            "kubernetes.config",
+            ('urllib3.exceptions', 'HTTPError')
+        ):
```

Comment:
Please parameterize this one to test one by one so that we sure the virtual raise if only one is missing